keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourcePersister.kt

```diff
@@ -17,27 +23,53 @@ import com.netflix.spinnaker.keel.plugin.supporting
 import org.slf4j.LoggerFactory
 import org.springframework.context.ApplicationEventPublisher
 import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Propagation.REQUIRED
+import org.springframework.transaction.annotation.Transactional
 import java.time.Clock
 
 @Component
 class ResourcePersister(
+  private val deliveryConfigRepository: DeliveryConfigRepository,
+  private val artifactRepository: ArtifactRepository,
   private val resourceRepository: ResourceRepository,
   private val handlers: List<ResolvableResourceHandler<*, *>>,
   private val clock: Clock,
   private val publisher: ApplicationEventPublisher
 ) {
-  fun upsert(resource: SubmittedResource<Any>): Resource<out Any> =
+  @Transactional(propagation = REQUIRED)
+  fun upsert(deliveryConfig: SubmittedDeliveryConfig): DeliveryConfig =
```

Comment:
This is the key method that needs to be transactional.

Reply:
`SqlDeliveryConfigRepository.store()` doesn't appear to be transaction though, so I feel like I'm missing something?

Reply:
It doesn't need to be to operate in the transaction that this method creates.

Reply:
I missed that spring-tx is managing transaction boundaries here 👍